build.env

```diff
@@ -17,7 +17,7 @@
 source ./tools/shell_functions.inc
 
 go version >/dev/null 2>&1 || fail "Go is not installed or is not in \$PATH. See https://vitess.io/contributing/build-from-source for install instructions."
-goversion_min 1.20.2 || fail "Go version reported: `go version`. Version 1.20.2+ required. See https://vitess.io/contributing/build-from-source for install instructions."
+goversion_min 1.20.2 || echo "Go version reported: `go version`. Version 1.20.2+ required. See https://vitess.io/contributing/build-from-source for install instructions."
```

Comment:
required -> recommended.
Otherwise the message is misleading, it's easy to assume that the build failed because historically that is what used to happen.

Reply:
Makes sense, i'll update this.

Reply:
Fix Pull Request: https://github.com/vitessio/vitess/pull/12841